Close the RichChoiceView dropdown when user clicks away

TW80000 opened the pull request:
This makes it behave more like native dropdowns. It also fixes an issue where two open `RichChoiceView`s would overlap if they were close enough to each other and one was above the other.

Commit:
Close the RichChoiceView dropdown when user clicks away

kgrgreer commented on src/foam/u2/view/RichChoiceView.js:
```diff
@@ -328,6 +328,26 @@ foam.CLASS({
+      // when the user clicks somewhere else.
+      var containerElement;
+      const fn = function(evt) {
+        var selfRect = (document.getElementById(self.id)).getClientRects()[0];
```
This is the same as this.el().getClientRects()[0]

kgrgreer commented on src/foam/u2/view/RichChoiceView.js:
```diff
@@ -328,6 +328,26 @@ foam.CLASS({
+      var containerElement;
+      const fn = function(evt) {
+        var selfRect = (document.getElementById(self.id)).getClientRects()[0];
+        var containerRect = (document.getElementById(containerElement.id)).getClientRects()[0];
```
see above

kgrgreer commented on src/foam/u2/view/RichChoiceView.js:
```diff
@@ -328,6 +328,26 @@ foam.CLASS({
+          self.isOpen_ = false;
+        }
+      };
+      window.addEventListener('click', fn);
```
Importing 'window' rather than using the global 'window' allows your app/view to work with multiple windows.

Commit:
Updates based on code review

Commit:
Don't select from the DAO in RichChoiceView until opened

We shouldn't waste resources by pulling from the DAO unless we know the
user is going to interact with the dropdown.

TW80000 commented:
@kgrgreer, all comments addressed, thanks for the review.

kgrgreer commented:
Very nice.
